tests/expected/intrinsics/size_of_dst.rs

```diff
@@ -0,0 +1,53 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+//! This test case checks the behavior of `size_of_val_raw` for dynamically sized types.
+
+#![feature(layout_for_ptr)]
+#![feature(ptr_metadata)]
+
+use std::mem::size_of_val_raw;
+use std::ptr;
+
+#[derive(Clone, Copy, kani::Arbitrary)]
+struct Wrapper<T: ?Sized> {
+    _size: usize,
+    _value: T,
+}
+
+#[kani::proof]
+pub fn check_size_of_adt_overflows() {
```

Comment:
Should we have tests for `align_of_val_raw` too?

Reply:
`align_of_val_raw` cannot overflow. It is actually safe today, but there's a question whether it will stay safe depending on how (see [this Zulip thread for more details](https://rust-lang.zulipchat.com/#narrow/channel/219381-t-libs/topic/.E2.9C.94.20Safety.20of.20.60size_of_val_raw.60.20and.20.60align_of_val_raw.60)). But I can add a few sanity checks too.

We could force alignment to be non-multiple of 2 by generating invalid vtable data for now. That itself may eventually become immediate UB, but it doesn't seem to be decided yet: https://github.com/rust-lang/unsafe-code-guidelines/issues/516.

Reply:
Huh, okay. Maybe just add a comment about how `align_of_val_raw` cannot overflow today, so we don't have tests? (Nothing too elaborate, just something so that a future reader realizes that this wasn't just an oversight on our part, but perhaps something that would need to change depending on whether it becomes unsafe in the future).
If the sanity checks aren't too much trouble, go ahead and add them, but I'd be fine with just a comment explaining the lack of tests.

Reply:
Actually, that is indeed UB. So we can't really test the invalid alignment case.